### DISCUSSION TITLE
KEDA's support for Dapr checkpointing

### DISCUSSION THREAD

**Discussion by tomkerkhove** [kedacore/keda#4320] [General]
Dapr v1.10 uses a new checkpointing behavior after migrating to [this SDK](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azeventhubs/README.md) in [this PR](https://github.com/dapr/components-contrib/pull/2453).

I'm assessing what that means for our Azure Event Hub scaler with https://github.com/berndverst & https://github.com/ItalyPaleAle from Dapr team.

Currently it looks like the `dapr` checkpoint strategy is no longer the best way, but I'm checking if we already have one that aligns with the Azure SDK.

**Comment by tomkerkhove** [comment]
Based on what @italypaleale shared the `default` checkpoint strategy should work.

@kedacore/keda-maintainers We might want to deprecated the `dapr` strategy in 2.10 since Dapr 1.10+ no longer uses that.

**Reply by ItalyPaleAle** [reply]
+1. Although please note that Dapr 1.9 will be supported until at least EOCY. Not sure what your support policy is, but it may help to keep it for those still on 1.9 or older

**Reply by JorTurFer** [reply]
If they have migrated to the new azure SDK, we can drop `dapr` support. I agree with deprecate it in v2.10. This gives to end users 6 months to upgrade their dapr instances and migrate the scaledobject. About having that strategy until dapr 1.9 is removed. When is the EOL? I don't have problems with extending the support some months if it's required

**Reply by tomkerkhove** [reply]
@JorTurFer As @ItalyPaleAle mentioned above, it is supported until end of 2023. I think it's OK to deprecate in 2.10 which will only remove it in 2.12.

Any thoughts on this @zroubalik @ahmelsayed @jeffhollan?

**Reply by zroubalik** [reply]
So we didn't hit 2.10. I think we can ~remove~ deprecate it in 2.11?

**Reply by tomkerkhove** [reply]
We need to deprecate it first, @zroubalik

**Reply by JorTurFer** [reply]
Let's deprecate it in next version

**Reply by zroubalik** [reply]
I meant to deprecate, of course.
